Remove warning from generator named base test

yuki24 opened the pull request:
This removes the following warning:

```
railties/test/generators/named_base_test.rb:9: warning: method redefined; discarding old pluralize_table_names
activesupport/lib/active_support/core_ext/class/attribute.rb:86: warning: previous definition of pluralize_table_names was here
railties/test/generators/named_base_test.rb:9: warning: method redefined; discarding old pluralize_table_names=
activesupport/lib/active_support/core_ext/class/attribute.rb:83: warning: previous definition of pluralize_table_names= was here
```

vipulnsward commented on railties/test/generators/named_base_test.rb:
```diff
@@ -6,6 +6,8 @@
+      undef pluralize_table_names
+      undef pluralize_table_names=
       attr_accessor :pluralize_table_names
     end
```
This looks like an overkill. wdyt?

kaspth replied:
Agreed. @yuki24 where is the original method defined?

yuki24 replied:
[here](https://github.com/rails/rails/blob/158c7eb1d61a28452e0aafd1e05314352eea2749/activerecord/lib/active_record/model_schema.rb#L50). Maybe we can just remove the attr_accessor in the test?

rafaelfranca replied:
> Maybe we can just remove the attr_accessor in the test?

I think it will be better.

kaspth replied:
One thing that worries me is the comment on line 5: "# Mock out what we need from AR::Base."
This file didn't expect the accessor to be defined (by a file being required).

It could just be that the files at the top now require some extra files, they didn't before. It's probably okay to remove it.

yuki24 committed:
Remove warning from generator named base test

This removes the following warning:

  railties/test/generators/named_base_test.rb:9: warning: method redefined; discarding old pluralize_table_names
  activesupport/lib/active_support/core_ext/class/attribute.rb:86: warning: previous definition of pluralize_table_names was here
  railties/test/generators/named_base_test.rb:9: warning: method redefined; discarding old pluralize_table_names=
  activesupport/lib/active_support/core_ext/class/attribute.rb:83: warning: previous definition of pluralize_table_names= was here

yuki24 commented:
seems like we can just remove `attr_accessor :pluralize_table_names`. Thanks for reviewing @vipulnsward @kaspth @rafaelfranca!

senny commented:
@yuki24 looks good. :yellow_heart:
